### ISSUE TITLE
Wrong injection on Named injections

### ISSUE THREAD

**Issue by sergiandreplace** [fabioCollini/DaggerMock#31]
This is quite a weird issue and we had a long time to determine what was going on.

This is the initial status:

1. I have several Strings injected via Provided. All them are using @Named in order to differentiate them. One of these injects is the base url
2. The base url is used in the builder of Retrofit:

```
@Provided
Retrofit provideRetrofit(@Named(URL_BASE) urlBase, other parameters...)
```
3. In the test we mock some objects, but not any of the named strings nor the retrofit.
4. In the test we define a constant
```
private static final MA_NAME = "ma_name"
```
5. the static test is used as response of a mocked dependency
given(someMock.getName).willReturn(MA_NAME)

So, when the test is run the MA_NAME string is injected to the provideRetrofit method instead of the @Named(URL_BASE) provided, making it crash because this is not an URL.

My guess is that the issue comes because using @Named or being an String. 

In the meantime, we removed the String injections transforming them into wrapper classes that provide the String.

**Comment by fabioCollini** [comment]
This issue can be fixed in two ways:

1. all fields are used, even the static ones that should be ignored

1. when a Named annotated object is found also the not annotated test fields are used. This was supposed to be a feature (see this test https://github.com/fabioCollini/DaggerMock/blob/master/lib/src/test/java/it/cosenonjaviste/daggermock/named/NamedTest.java) but maybe this was a bad decision

In the days I am going to fix it and release a new version.
Thanks for your report!

**Comment by fabioCollini** [comment]
Hi, I have committed the fixes, can you try to use the version b034297429 to verify that it fixes your issue?
Thanks again

**Comment by sergiandreplace** [comment]
I'm quite busy these days and test it will take me a time. I'll keep you posted.

**Comment by caefree** [comment]
We are also having this same issue..
Is there an easier fix, on the client side?

**Comment by fabioCollini** [comment]
The fix of this issue is already available in version 0.6.4. Which version are you using? Do you still get the same error?

**Comment by caefree** [comment]
I'm on 0.6.2, will update to 0.6.4 and let you know asap.
